models/userStatus.js

```diff
@@ -193,4 +200,55 @@ const updateAllUserStatus = async () => {
   }
 };
 
-module.exports = { deleteUserStatus, getUserStatus, getAllUserStatus, updateUserStatus, updateAllUserStatus };
+/**
+ * Cancels the Out-of-Office (OOO) status for a user.
+ * @param {string} userId - The ID of the user.
+ * @returns {Promise<object>} - A promise that resolves to an object with cancellation details.
+ * @throws {Error} - If there is an error fetching the user status document or updating the status.
+ */
+
+const cancelOooStatus = async (userId) => {
+  try {
+    let userStatusDoc;
+    let isActive;
+    try {
+      userStatusDoc = await userStatusModel.where("userId", "==", userId).get();
+    } catch (error) {
+      logger.error(`Unable to fetch user status document from the firestore : ${error.message}`);
+      throw error;
+    }
+    if (!userStatusDoc.size) {
+      throw new NotFound("No User status document found");
+    }
+    const [userStatusDocument] = userStatusDoc.docs;
+    const docId = userStatusDocument.id;
+    const { futureStatus, ...docData } = userStatusDocument.data();
+    if (docData.currentStatus.state !== userState.OOO) {
+      throw new Forbidden(
+        `The ${userState.OOO} Status cannot be canceled because the current status is ${docData.currentStatus.state}.`
+      );
+    }
+    try {
+      isActive = await checkIfUserHasLiveTasks(userId, tasksModel);
+    } catch (error) {
+      logger.error(`Unable to fetch user status based on the task : ${error.message}`);
+      throw error;
```

Comment:
not caught in tests please add coverage

Reply:
This will happen when the call to Firestore DB fails. I am not sure how to simulate that 😢

Reply:
Done